docs: [WIP] Update routing/request/response docs (#2912)
litestar-org/litestar: docs/usage/routing/index.rst

Routing
=======

Litestar's routing system is based around a few basic components:

Route handlers
    Callables that receive parameters from the request or dependencies and return
    response data

Controllers
    Classes that group together multiple route handlers and define common attributes
    such as dependencies and middlewares

Routers
    A lightweight application on which route handlers, controllers and other routers can
    be registered, as well as common configurations such as dependencies and middlewares

Applications

>     Instances of :class:`~litestar.app.Litestar`, a special type of router that serves
>     as the entry point of an application, in addition to receive the same configuration
>     routers can

Reviewer: ```suggestion
    Instances of :class:`~litestar.app.Litestar`, the entry point of an application, and a 
    specialized type of router, so it can receive the same configuration as routers can
```
This wording was a bit awkward - not sure if my suggestion is much better though!

    :language: python

When not provided, the path defaults to ``/``.

Routers
+++++++

The path on a router can be set via the ``path`` argument. Unlike handlers and
controllers, specifying a path is not optional.

.. literalinclude:: /examples/routing/router_path.py
    :language: python

Setting a base path for the application
++++++++++++++++++++++++++++++++++++++++

Sometimes it is desirable to set a base path for the entire application. While it is
generally advisable to handle this outside the application itself (e.g. via a reverse
proxy), this can be achieved by registering all route handlers with a router, and
setting the base path there.

Dynamic paths with path parameters
++++++++++++++++++++++++++++++++++

Dynamic paths can be created with the help of path parameters. These are part of the
path string and follow the basic form of ``{<parameter name>:<parameter type>}``, where
``<parameter name>`` is the name of the parameter used to make its value available to
the route handler and what will be represented in the OpenAPI schema, and
``<parameter type>`` a basic Python type that this parameter will be validated against

> and, if possible, converted to.

Reviewer: > These are part of the
path string and follow the basic form of ``{<parameter name>:<parameter type>}``, where
``<parameter name>`` is the name of the parameter used to make its value available to
the route handler and what will be represented in the OpenAPI schema, and
``<parameter type>`` a basic Python type that this parameter will be validated against
and, if possible, converted to.

This is quite a big sentence - might be better to break it up into smaller ideas, e.g., a sentence that introduces the two components of the parameter, then one explaining parameter name, and another to explain parameter type.
